bin/p2-dsctl/main.go

```diff
@@ -0,0 +1,237 @@
+			}
+			ds.Disabled = false
+			return ds, nil
+		}
+
+		_, err := dsstore.MutateDS(id, mutator)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		fmt.Printf("The daemon set '%s' has been successfully enabled in consul", id.String())
+		fmt.Println()
+
+	case CmdDisable:
+		id := ds_fields.ID(*disableID)
+
+		mutator := func(ds ds_fields.DaemonSet) (ds_fields.DaemonSet, error) {
+			if ds.Disabled {
+				return ds, util.Errorf("Daemon set has already been enabled")
+			}
+			ds.Disabled = true
+			return ds, nil
+		}
+
+		_, err := dsstore.MutateDS(id, mutator)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		fmt.Printf("The daemon set '%s' has been successfully disabled in consul", id.String())
+		fmt.Println()
+
+	case CmdDelete:
+		id := ds_fields.ID(*deleteID)
+		err := dsstore.Delete(id)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		fmt.Printf("The daemon set '%s' has been successfully deleted from consul", id.String())
+		fmt.Println()
+
+	case CmdUpdate:
+		id := ds_fields.ID(*updateID)
+
+		mutator := func(ds ds_fields.DaemonSet) (ds_fields.DaemonSet, error) {
+			changed := false
+
+			if *updateAZ != "" {
+				selector := selectorFrom(pc_fields.AvailabilityZone(*updateAZ))
+				if ds.NodeSelector.String() != selector.String() {
+					changed = true
+					ds.NodeSelector = selectorFrom(pc_fields.AvailabilityZone(*updateAZ))
+				}
+			}
+			if *updateMinHealth != "" {
+				minHealth, err := strconv.Atoi(*updateMinHealth)
+				if err != nil {
+					log.Fatalf("Invalid value for minimum health, expected integer")
+				}
+				if ds.MinHealth != minHealth {
+					changed = true
+					ds.MinHealth = minHealth
+				}
+			}
+			if *updateName != "" {
+				name := ds_fields.ClusterName(*updateName)
+				if ds.Name != name {
+					changed = true
+					ds.Name = name
+				}
+			}
+			if *updatePodID != "" {
+				podID := types.PodID(*updatePodID)
+				if ds.PodID != podID {
+					changed = true
+					ds.PodID = podID
+				}
+			}
+			if *updateManifest != "" {
+				manifest, err := pods.ManifestFromPath(*updateManifest)
+				if err != nil {
+					return ds, util.Errorf("%s", err)
+				}
+				dsSHA, err := ds.Manifest.SHA()
+				if err != nil {
+					return ds, util.Errorf("Unable to get SHA from consul daemon set manifest: %v", err)
+				}
+				newSHA, err := manifest.SHA()
+				if err != nil {
+					return ds, util.Errorf("Unable to get SHA from new manifest: %v", err)
+				}
+				if dsSHA != newSHA {
+					changed = true
+					ds.Manifest = manifest
+				}
+			}
+
+			if !changed {
+				return ds, util.Errorf("No changes were made")
+			}
+			return ds, nil
+		}
+
+		_, err := dsstore.MutateDS(id, mutator)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		fmt.Printf("The daemon set '%s' has been successfully updated in consul", id.String())
+		fmt.Println()
+
+	default:
+		log.Fatalf("Unrecognized command %v", cmd)
+	}
+}
+
+func selectorFrom(az pc_fields.AvailabilityZone) labels.Selector {
```

Comment:
side note: it's interesting to me that availability zone is expected to be in the node selector but is not present on the daemon set itself, the former suggests that you'd want a daemon set to span more than a single availability zone, so maybe you want to support a node selector using `labels.InOperator` to support selecting nodes from multiple availability zones. Not a blocker for this PR though, we'll watch what use cases users have with daemon sets.